src/AppInstallerCLICore/Commands/InstallCommand.cpp

```diff
@@ -143,12 +145,28 @@ namespace AppInstaller::CLI
     {
         context.SetFlags(ContextFlag::ShowSearchResultsOnPartialFailure);
 
-        context <<
-            Workflow::ReportExecutionStage(ExecutionStage::Discovery) <<
-            Workflow::GetManifest <<
-            Workflow::SelectInstaller <<
-            Workflow::EnsureApplicableInstaller <<
-            Workflow::CheckForUnsupportedArgs <<
-            Workflow::InstallSinglePackage;
+        if (context.Args.Contains(Execution::Args::Type::Manifest))
+        {
+            context <<
+                Workflow::ReportExecutionStage(ExecutionStage::Discovery) <<
+                Workflow::GetManifestFromArg <<
+                Workflow::SelectInstaller <<
+                Workflow::EnsureApplicableInstaller <<
+                Workflow::InstallSinglePackage;
+        }
+        else
+        {
+            context <<
+                Workflow::ReportExecutionStage(ExecutionStage::Discovery) <<
+                Workflow::OpenSource();
+
+            if (!context.Args.Contains(Execution::Args::Type::Force))
+            {
+                context <<
+                    Workflow::OpenCompositeSource(Repository::PredefinedSource::Installed, false, Repository::CompositeSearchBehavior::AvailablePackages);
+            }
+
+            context << Workflow::InstallOrUpgradeSinglePackage(false);

```

Comment:
Does this also need to be applied to `import`?  More specifically, does `import` need to be calling `InstallOrUpgradeSinglePackage`?